### PR TITLE
Add new setting

### PR DESCRIPTION
**Pull request by DavidFeldhoff**
Add new setting  `azdevops-vscode-simplify.createBranch.branchNameProposal` which enables also adding the work item description and obsolete `useWorkitemIdInBranchName` setting for it.
Remove any non-word-characters from the work item description, make it lowercase, replace spaces with dashes and cut at 370 characters as the limit is somewhere between 380-390 (read different values in different forums, but 370 should be save and enough)

### CODE REVIEW

**Comment by DavidFeldhoff** [comment]
Linking #36 to this PR

File: src/api/azdevops-api.ts [hunk 1]
```diff
@@ -692,4 +691,26 @@ export class WorkItemTreeItem extends vscode.TreeItem {
             return remoteRefs;
         }
     }
+
+    private getBranchNameProposal() {
+        const gitPrefix = vscode.workspace.getConfiguration('git').get('branchPrefix', "");
+        let branchNameProposal = gitPrefix !== "" ? `${gitPrefix}` : "";
+        let safeDescription = this.label.trim().toLowerCase().replace(/ /g, "-");
```
Comment (tfenster) [inline src/api/azdevops-api.ts:698]: should we also guard against special characters? I would probably just take letters and numbers and replace everything else with a dash. If that creates a weird name, the user can always change the proposal

File: src/helpers.ts [hunk 2]
```diff
@@ -23,14 +23,30 @@ export function showWorkItemTypes(): string[] {
...
-export function useWorkitemIdInBranchName(): boolean {
-    return vscode.workspace.getConfiguration('azdevops-vscode-simplify').get('useWorkitemIdInBranchName', false) ||
-        vscode.workspace.getConfiguration('azdevops-vscode-simplify').get('createBranch.useWorkitemIdInBranchName', false);
-}
-export function createBranchBasedOn(): string{
+export function getBranchNameProposalSetting(): BranchNameProposal {
+    let branchNameProposal = BranchNameProposal.nothing;
+    const branchNameProposalAsString = vscode.workspace.getConfiguration('azdevops-vscode-simplify').get<string>('createBranch.branchNameProposal');
+    if (branchNameProposalAsString) {
+        branchNameProposal = BranchNameProposal[branchNameProposalAsString as any] as unknown as BranchNameProposal;
```
Comment (tfenster) [inline src/helpers.ts:30]: isn't TypeScript great 😆

**Review by tfenster — CHANGES_REQUESTED** [review]

File: src/api/azdevops-api.ts (hunk 1 above)
Comment (DavidFeldhoff) [reply src/api/azdevops-api.ts:698]: Isn't that approximately what I'm doing? I'm replacing first spaces with a dash and then remove everything that's not a word character or a dash.🤔

File: src/helpers.ts (hunk 2 above)
Comment (DavidFeldhoff) [reply src/helpers.ts:30]: Yes, but.. no 🤣 NOT this time 😅

File: src/api/azdevops-api.ts (hunk 1 above)
Comment (tfenster) [reply src/api/azdevops-api.ts:698]: absolutely. Sorry, I only saw the first replace and failed to note the second one

**Review by tfenster — APPROVED** [review]

**Pull request merged**